src/v/kafka/protocol/response_writer.h

```diff
@@ -347,6 +346,15 @@ class response_writer {
     // compliant
     uint32_t write_tags() { return write_unsigned_varint(0); }
 
+    /// Used when a size (in bytes) must be prepended before serializing the
+    /// object itself. So far only used for encoding tag headers.
+    uint32_t write_size_prepended(iobuf&& buf) {
+        const auto buf_sz = buf.size_bytes();
+        const auto sz = write_unsigned_varint(buf.size_bytes());
+        _out->append(std::move(buf));
+        return _out->size_bytes() - (sz + buf_sz);
```

Comment:
Nice helper func. Thanks for the ASCII-art diagram of the tag encoding.. made it easy to understand this stuff.